### PR TITLE
completed up to highestscorefrom

### PR DESCRIPTION
**Pull request by griffifam**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? |  Certain built in methods were the same as in ruby. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | Yes I use W3schools to get a visual of how things worked and examples. 
What was a challenge you were able to overcome on this assignment? | Understanding how to call a function within a function
What is your favorite thing about learning a new programming language? | I enjoy learning new functionality; like being able to call a function on a specific element in HTML
What is your least favorite thing about learning a new programming language? | Not knowing what I don't know.
Do you have any recommendations on how we could improve this project for the next cohort? | :)

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x
**Code Requirements** |
`drawLetters` method |  x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic |
All tests for `highestScoreFrom` pass |
**Overall** |

Nice work on this, Amber Lynn! I like the solutions you made; you did a great job of iterating through and using conditional logic to work through the project. Unfortunately, a lot of the intricate tie-breaking logic in the function `highestScoreFrom` doesn't pass, though, and wasn't implemented.

That being said, I like your code style, and it looks good to me. I have only a few comments; otherwise I think you did a great job. It's also worth noting that none of my comments are about syntax: You had perfect syntax throughout the entire project :)

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,136 @@
...
+      return true;
+    }
+    return false;
   },
+  scoreWord( word ) {
+    let hand = word.toUpperCase();
+
+    if ( hand == "" || hand == " ") {
+      return 0;
+    }
```
Comment (tildeee) [inline src/adagrams.js:101]: when will this ever be true? in what cases will `hand` be an empty string?

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,136 @@
...
+      scoreHash[word] = Adagrams.scoreWord( word );
+    });
+
+    for( let key in scoreHash ) {
+      if (scoreHash[key] > highest_value) {
+        highest_value = scoreHash[key];
+        highest_key = key;
+      }
+    }
+    return { word: `${highest_key}`, score: Number(`${highest_value}`) };
```
Comment (tildeee) [inline src/adagrams.js:131]: Here, you interpolate the variables `highest_key` and `highest_value`. You don't need to interpolate them: you can just do the following:

`return { word: highest_key, score: highest_value }`
